### PR TITLE
[CIT-515] fix the timout issue

### PR DESCRIPTION
**Pull request by PradeepKiruvale**
## Proposed changes

Fix the timeout issue with the mqtt test. The sleep must be created outside the select call.

## Types of changes

What types of changes does your code introduce to `thin-edge.io`?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactorings that doesn't explicitly fix a bug or add any new functionality)
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [ ] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [ ] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: common/mqtt_client/tests/mqtt_pub_sub_test.rs [hunk 1]
```diff
@@ -24,17 +24,21 @@ fn sending_and_receiving_a_message() {
...
+        tokio::pin!(sleep);
 
-        tokio::select! {
-            msg = received.next() => Ok(msg),
-            _ = sleep(Duration::from_millis(1000)) => Ok(None)
+        loop {
+            tokio::select! {
+                msg = received.next() => { mqtt_server_handle.abort(); return Ok(msg);},
+                _ = &mut sleep => {mqtt_server_handle.abort(); return Ok(None);}
+            }
```
Comment (didier-wenzek) [inline common/mqtt_client/tests/mqtt_pub_sub_test.rs:34]: * What problem do you try to solve?
* Why a loop with a return after the first iteration in any cases?
* What is the difference with the sleep started in the select?

**Review by didier-wenzek — CHANGES_REQUESTED** [review]
What problem do you try to solve?

File: common/mqtt_client/tests/mqtt_pub_sub_test.rs (hunk 1 above)
Comment (PradeepKiruvale) [reply common/mqtt_client/tests/mqtt_pub_sub_test.rs:34]: Yes, I tried fixing the issue by correct way of creating the timer inside the select statement. As Lukasz mentioned it worked for me for 2/3 tries. I tried the same test with mosquito broker, tests work properly even if I run them for hours continuously.

As per as creation of the sleep and pinning outside the select loop is required when the sleep is used multiple times.
Can find more about it here https://docs.rs/tokio/1.10.0/tokio/time/struct.Sleep.html

**Comment by makr11st** [comment]
Closing as this has now been moved over to  #389

**Pull request closed without merging**